cmip6_downscaling/methods/gard.py

```diff
@@ -88,134 +91,86 @@ def gard_fit_and_predict(
     return out
 
 
-def calc_correlation_length_scale(
-    data: xr.DataArray,
-    seasonality_period: int = 31,
-    temporal_scaler: float = 1000.0,
-) -> Dict[str, float]:
-    """
-    find the correlation length for a dataarray with dimensions lat, lon and time
-    it is assumed that the correlation length in lat and lon directions are the same due to the implementation in gstools
-    """
-    for dim in ['lat', 'lon', 'time']:
-        assert dim in data.dims
-
-    # remove seasonality before finding correlation length, otherwise the seasonality correlation dominates
-    seasonality = (
-        data.rolling({'time': seasonality_period}, center=True, min_periods=1)
-        .mean()
-        .groupby('time.dayofyear')
-        .mean()
-    )
-    detrended = data.groupby("time.dayofyear") - seasonality
-
-    # find spatial length scale
-    fields = detrended.values
-    bin_center, gamma = gs.vario_estimate(
-        # TODO: need to figure whether we need to do .values for this function
-        pos=(detrended.lon.values, detrended.lat.values),
-        field=fields,
-        latlon=True,
-        mesh_type='structured',
-    )
-    spatial = gs.Gaussian(dim=2, latlon=True, rescale=gs.EARTH_RADIUS)
-    spatial.fit_variogram(bin_center, gamma, sill=np.mean(np.var(fields, axis=(1, 2))))
-
-    # find temporal length scale
-    # break the time series into fields of 1 year length, since otherwise the algorithm struggles to find the correct length scale
-    fields = []
-    day_in_year = 365
-    for yr, group in detrended.groupby('time.year'):
-        # TODO: this is a very long list for a large domain, perhaps need random sampling
-        v = (
-            group.isel(time=slice(0, day_in_year))
-            .stack(point=['lat', 'lon'])
-            .transpose('point', 'time')
-            .values
-        )
-        fields.extend(list(v))
-    t = np.arange(day_in_year) / temporal_scaler
-    bin_center, gamma = gs.vario_estimate(pos=t, field=fields, mesh_type='structured')
-    temporal = gs.Gaussian(dim=1)
-    temporal.fit_variogram(bin_center, gamma, sill=np.mean(np.var(fields, axis=1)))
-
-    return {'temporal': temporal.len_scale, 'spatial': spatial.len_scale}
-
-
-def generate_scrf(
-    data: xr.Dataset,
+def read_scrf(
```

Comment:
i assume by shifting the `calc_correlation_length_scale` and `write_scrf` bits to the notebook the code is identical? like no tweaks along the way that we'd want to keep track of?

Reply:
oh there are def. tweaks